File: app/src/main/java/com/example/calculator/activities/MainActivity.java
package com.example.calculator.activities;

import android.annotation.SuppressLint;
import android.content.Intent;
import android.os.Bundle;
import android.view.Menu;
import android.view.MenuItem;

import androidx.annotation.NonNull;
import androidx.appcompat.app.AppCompatActivity;

import com.example.calculator.NonSwipeableViewPager;
import com.example.calculator.R;
import com.example.calculator.adpters.ViewPagerAdapter;
import com.example.calculator.fragments.CalculatorFragment;
import com.example.calculator.fragments.ConverterFragment;
import com.google.android.material.tabs.TabLayout;

import java.util.Objects;

public class MainActivity extends AppCompatActivity {
    TabLayout tabLayout;
    NonSwipeableViewPager viewPager;

    @SuppressLint("ClickableViewAccessibility")
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        setSupportActionBar(findViewById(R.id.topAppBar));
        tabLayout = findViewById(R.id.tabLayout);
        viewPager = findViewById(R.id.pager);
        ViewPagerAdapter pagerAdapter = new ViewPagerAdapter(getSupportFragmentManager(), 1);
        pagerAdapter.addFragment(new CalculatorFragment());
        pagerAdapter.addFragment(new ConverterFragment());
        viewPager.setAdapter(pagerAdapter);
        tabLayout.setupWithViewPager(this.viewPager);
        Objects.requireNonNull(this.tabLayout.getTabAt(0)).setIcon(R.drawable.ic_filled_calculate);
        Objects.requireNonNull(this.tabLayout.getTabAt(1)).setIcon(R.drawable.ic_outline_converter);
        this.tabLayout.addOnTabSelectedListener(new TabLayout.OnTabSelectedListener() {
            public void onTabSelected(TabLayout.Tab tab) {
                if (tab.getPosition() == 0) {
                    tab.setIcon(R.drawable.ic_filled_calculate);
                } else if (tab.getPosition() == 1) {
                    tab.setIcon(R.drawable.ic_filled_converter);
                }
            }

            public void onTabUnselected(TabLayout.Tab tab) {
                if (tab.getPosition() == 0) {
                    tab.setIcon(R.drawable.ic_outline_calculator);
                } else if (tab.getPosition() == 1) {
                    tab.setIcon(R.drawable.ic_outline_converter);
                }
            }

            public void onTabReselected(TabLayout.Tab tab) {
            }
        });
    }

    @Override
    public boolean onCreateOptionsMenu(Menu menu) {
        getMenuInflater().inflate(R.menu.app_menu, menu);
        return super.onCreateOptionsMenu(menu);
    }

    @Override
    public boolean onOptionsItemSelected(@NonNull MenuItem item) {
        if (item.getItemId() == R.id.settingButton) {
            startActivity(new Intent(this, SettingsActivity.class));
        }
        return super.onOptionsItemSelected(item);
    }
}

File: app/src/main/java/com/inator/calculator/activities/MainActivity.kt
package com.inator.calculator.activities

import android.content.Intent
import android.os.Bundle
import android.view.Menu
import android.view.MenuItem
import android.view.inputmethod.InputMethodManager
import androidx.activity.viewModels
import androidx.appcompat.app.AppCompatActivity
import androidx.core.content.getSystemService
import com.google.android.material.dialog.MaterialAlertDialogBuilder
import com.google.android.material.tabs.TabLayout
import com.google.android.material.tabs.TabLayout.OnTabSelectedListener
import com.google.android.material.tabs.TabLayoutMediator
import com.inator.calculator.R
import com.inator.calculator.adapters.ViewPagerAdapter
import com.inator.calculator.databinding.ActivityMainBinding
import com.inator.calculator.extras.ZoomOutPageTransformer
import com.inator.calculator.fragments.CalculatorFragment
import com.inator.calculator.fragments.ConverterFragment
import com.inator.calculator.fragments.CurrencyFragment
import com.inator.calculator.viewmodel.HistoryViewModel


class MainActivity : AppCompatActivity() {

    private val historyViewModel: HistoryViewModel by viewModels()
    private lateinit var binding: ActivityMainBinding

    val topAppBar get() = binding.topAppBar
    val historyBar get() = binding.historyBar

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setTheme(R.style.Theme_Calculator)
        binding = ActivityMainBinding.inflate(layoutInflater)
        setContentView(binding.root)
        setSupportActionBar(binding.topAppBar)
        setUpViews()
    }

    private fun setUpViews() {
        val pagerAdapter = ViewPagerAdapter(this)
        pagerAdapter.apply {
            addFragment(CalculatorFragment())
            addFragment(ConverterFragment())
            addFragment(CurrencyFragment())
        }
        binding.pager.apply {
            adapter = pagerAdapter
            offscreenPageLimit = 2
            setPageTransformer(ZoomOutPageTransformer())
            isUserInputEnabled = false
        }

        binding.tabLayout.apply {
            TabLayoutMediator(
                this, binding.pager
            ) { _: TabLayout.Tab, _: Int -> }.attach()

            getTabAt(0)?.setIcon(R.drawable.ic_calculator_filled)
            getTabAt(1)?.setIcon(R.drawable.ic_converter_outline)
            getTabAt(2)?.setIcon(R.drawable.ic_currency_outline)

            addOnTabSelectedListener(object : OnTabSelectedListener {

                override fun onTabSelected(tab: TabLayout.Tab) {
                    when (tab.position) {
                        0 -> {
                            tab.setIcon(R.drawable.ic_calculator_filled)
                            // Hide keyboard
                            getSystemService<InputMethodManager>()?.hideSoftInputFromWindow(
                                this@apply.windowToken,
                                0
                            )
                        }

                        1 -> tab.setIcon(R.drawable.ic_converter_filled)

                        2 -> tab.setIcon(R.drawable.ic_currency_filled)
                    }
                }

                override fun onTabUnselected(tab: TabLayout.Tab) {
                    tab.setIcon(
                        when (tab.position) {
                            0 -> R.drawable.ic_calculator_outline

                            1 -> R.drawable.ic_converter_outline

                            2 -> R.drawable.ic_currency_outline

                            else -> R.drawable.ic_calculator_outline
                        }
                    )
                }

                override fun onTabReselected(tab: TabLayout.Tab?) {}
            })
        }
    }

    override fun onCreateOptionsMenu(menu: Menu): Boolean {
        menuInflater.inflate(R.menu.app_menu, menu)
        return super.onCreateOptionsMenu(menu)
    }

    override fun onOptionsItemSelected(item: MenuItem): Boolean {
        when (item.itemId) {
            R.id.action_settings -> {
                startActivity(Intent(this, SettingsActivity::class.java))
            }
            R.id.delete_history -> {
                showDeleteDialog()
            }
        }
        return super.onOptionsItemSelected(item)
    }

    private fun showDeleteDialog() {
        MaterialAlertDialogBuilder(this)
            .setMessage(resources.getString(R.string.delete_message))
            .setTitle(resources.getString(R.string.delete_title))
            .setIcon(R.drawable.ic_delete_history)
            .setPositiveButton(android.R.string.ok) { _, _ ->
                historyViewModel.deleteAllHistory()
            }
            .setNegativeButton(R.string.cancel) { dialog, _ ->
                dialog.dismiss()
            }
            .create()
            .show()
    }
}